daemon/mgr/image.go

```diff
@@ -268,16 +253,15 @@ func (c *imageCache) get(idOrRef string) (*types.ImageInfo, error) {
 		return nil, err
 	}
 
-	var id string
+	id := idOrRef
 	if refDigest, ok := refNamed.(reference.Digested); ok {
 		id = c.refToID[refDigest.String()]
-	} else {
-		refTagged := reference.WithDefaultTagIfMissing(refNamed).(reference.Tagged)
-		if len(c.refToID[refTagged.String()]) == 0 {
-			id = idOrRef
-		} else {
-			id = c.refToID[refTagged.String()]
-		}
+	}
+	if refTagged, ok := refNamed.(reference.Tagged); ok {
+		id = c.refToID[refTagged.String()]
+	}
+	if id == "" {
+		return nil, errors.Wrap(errtypes.ErrNotfound, "image: "+id)
```

Comment:
error msg is incomplete while using repoDigest，maybe we should raise error on line 258.
`Error: failed to run container: {"message":"image: : not found"}`